### PR TITLE
Improve Update Container

### PR DESCRIPTION
**Pull request by cedelavergne-ledger**
This PR improve the following:
- Add setting to keep Terminal Window opened after a successful _Container Update_
- Add setting for the max number of _Container Update_ retries
- Add a new `containerStatus` function to clean the code
- Hide Terminal Window when the _Container Update_ is done
- Only retry the _Container Update_ if number of retries < max value from settings
- Change version to 0.1.6

This PR also fixes https://github.com/LedgerHQ/ledger-vscode-extension/issues/2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,6 +101,16 @@
                 "default": {"app-boilerplate": "apk add gcc musl-dev python3-dev"},
                 "scope": "application",
                 "description": "Additional dependencies to install for each app."
+              },
+              "ledgerDevTools.keepContainerTerminal": {
+                "type": "boolean",
+                "default": true,
+                "markdownDescription": "Keep Terminal window opened when Container Update successfully."
```
Comment (agrojean-ledger) [inline package.json:108]: Reword : `Keep the Terminal window opened after a successful Container Update.`

File: src/containerManager.ts [hunk 2]
```diff
@@ -32,39 +32,49 @@ export class ContainerManager {
...
         }
-      } else {
-        this.statusBarManager.updateDevImageItem(DevImageStatus.stopped);
-        this.treeProvider.updateContainerLabel(DevImageStatus.stopped);
       }
+      return false;
+
     } catch (error: any) {
       console.log(`Docker error : ${error.message}`);
-      this.statusBarManager.updateDevImageItem(DevImageStatus.stopped);
```
Comment (agrojean-ledger) [inline src/containerManager.ts:66]: It was useless to have the status bar and tree item updates here (in the catch part) ?

File: src/extension.ts [hunk 3]
```diff
@@ -68,10 +68,29 @@ export function activate(context: vscode.ExtensionContext) {
     }
   });
 
+  let nbUpdate: number = 0;
   vscode.tasks.onDidEndTask((event) => {
     const taskName = event.execution.task.name;
     if (taskName.startsWith("Update Container")) {
-      containerManager.manageContainer();
+      if (containerManager.containerStatus() == true) {
```
Comment (agrojean-ledger) [inline src/extension.ts:75]: I suggest we create a method of `containerManager` that will perform the stuff you've added (the retries etc) so all the container logic stays in the same class

File: src/containerManager.ts [hunk 4]
```diff
@@ -32,39 +32,49 @@ export class ContainerManager {
...
+      if (currentApp) {
+        this.statusBarManager.updateDevImageItem(DevImageStatus.stopped);
+        this.treeProvider.updateContainerLabel(DevImageStatus.stopped);
+        console.log(`Ledger: Container ${currentApp.containerName} not found, respawning it.`);
+        this.taskProvider.executeTaskByName("Update Container");
+      }
+    }
+  }
+
+  containerStatus(): boolean {
```
Comment (agrojean-ledger) [inline src/containerManager.ts:46]: Can't we get something a bit more explicit than a boolean as a return value here ?

File: README.md [hunk 5]
```diff
@@ -47,9 +47,16 @@ This extension contributes the following settings:
 * `ledgerDevTools.onboardingSeed`: Set the device quick onboarding 24-word Seed phrase.
 * `ledgerDevTools.dockerImage`: Set the Ledger developer tools Docker image.
 * `ledgerDevTools.additionalDepsPerApp`: Add dependencies for current app's functional tests (for instance 'apk add python3-protobuf').
+* `ledgerDevTools.keepContainerTerminal`: Indicates to keep the Terminal window opened after a successful Container Update.
+* `ledgerDevTools.ContainerUpdateTentatives`: Set the max number of Container Update tentatives.
```
Comment (agrojean-ledger) [inline README.md:51]: I would reword this setting to `ContainerUpdateRetries` (tentative does not have the proper meaning in english)

**Comment by agrojean-ledger** [comment]
@cedelavergne-ledger as this PR does not change any host command executed, we should be fine with not testing on all supported platforms (no need to test on windows or mac, it should still work)

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,17 @@ All notable changes to this project will be documented in this file.
...
+## [0.1.6]
+
+### Added
+
+    - Add setting parameter allowing to automatically close the terminal Window when the Container Update is successful.
+    - Add setting parameter to set the max number of allowed Container Update retries.
+
+### Fixed
+
+    - Infinite loops whn the Container Update fails.
```
Comment (agrojean-ledger) [inline CHANGELOG.md:17]: whn > when

File: package.json (hunk 1 above)
Comment (cedelavergne-ledger) [reply package.json:108]: Done

File: README.md (hunk 5 above)
Comment (cedelavergne-ledger) [reply README.md:51]: Done

File: CHANGELOG.md (hunk 6 above)
Comment (cedelavergne-ledger) [reply CHANGELOG.md:17]: Done

File: src/extension.ts (hunk 3 above)
Comment (cedelavergne-ledger) [reply src/extension.ts]: Done

File: src/containerManager.ts (hunk 4 above)
Comment (cedelavergne-ledger) [reply src/containerManager.ts]: Done

File: src/containerManager.ts [hunk 7]
```diff
@@ -31,40 +33,72 @@ export class ContainerManager {
...
+      if (currentApp) {
+        this.statusBarManager.updateDevImageItem(DevImageStatus.stopped);
+        this.treeProvider.updateContainerLabel(DevImageStatus.stopped);
+        console.log(`Ledger: Container ${currentApp.containerName} not found, respawning it.`);
+        this.taskProvider.executeTaskByName("Update Container");
+      }
+    }
+  }
+
+  checkRetries(): void {
```
Comment (agrojean-ledger) [inline src/containerManager.ts:83]: @cedelavergne-ledger maybe change the name to something a little bit more explicit like `checkUpdateRetries`

**Review by agrojean-ledger — APPROVED** [review]

File: src/containerManager.ts (hunk 7 above)
Comment (cedelavergne-ledger) [reply src/containerManager.ts]: Done!
